app/src/main/java/ch/epfl/sdp/appart/utils/PermissionRequest.java

```diff
@@ -37,9 +37,67 @@ public static void askForLocationPermission(Activity activity, Runnable permissi
                 //Ask permission
                 requestPermissionLauncher.launch(new String[]{Manifest.permission.ACCESS_COARSE_LOCATION, Manifest.permission.ACCESS_FINE_LOCATION});
             }
+        } else {
+            permissionGranted.run();
+        }
+    }
+
+    public static void askForCameraPermission(Activity activity, Runnable permissionGranted,
+                                              Runnable permissionRefused) {
+        ActivityResultLauncher<String[]> requestPermissionLauncher;
+        String[] permissions;
+        
+        if (Build.VERSION.SDK_INT < 29) {
+            permissions = new String[]{Manifest.permission.READ_EXTERNAL_STORAGE,
+                    Manifest.permission.WRITE_EXTERNAL_STORAGE,
+                    Manifest.permission.CAMERA};
+        } else {
+            permissions = new String[]{
+                    Manifest.permission.CAMERA};
         }
-        else {
+
+        requestPermissionLauncher = resultLauncherFor(activity, permissionGranted,
+                permissionRefused, permissions);
+
+        if (!permissionsAlreadyGranted(activity, permissions)) {
+            requestPermissionLauncher.launch(permissions);
+        } else {
             permissionGranted.run();
         }
     }
+
+    /**
+     * Returns an ActivityResultLauncher for the given permissions requests
+     */
+    private static ActivityResultLauncher<String[]> resultLauncherFor(Activity activity,
+                                                                      Runnable permissionGranted,
+                                                                      Runnable permissionRefused,
+                                                                      String... permissions) {
+        return ((ComponentActivity) activity).registerForActivityResult(
+                new ActivityResultContracts.RequestMultiplePermissions(), isGranted -> {
+                    Boolean everythingGranted = true;
+                    for (int i = 0; i < permissions.length; i++) {
+                        if (!isGranted.get(permissions[i])) everythingGranted = false;
+                    }
+                    if (everythingGranted) {
+                        permissionGranted.run();
+                    } else {
+                        permissionRefused.run();
+                    }
+                });
+    }
+
+    /**
+     * Checks if the given permissions are granted, if not ask for them
+     */
+    private static boolean permissionsAlreadyGranted(Activity activity, String... permissions) {
+        Boolean everythingGranted = true;
+        for (int i = 0; i < permissions.length; i++) {
+            if (ActivityCompat.checkSelfPermission(activity, permissions[i])
+                    != PackageManager.PERMISSION_GRANTED) {
+                everythingGranted = false;
+            }
+        }
+        return everythingGranted;
+    }
```

Comment:
I think I made something like in my Local DB PR. I guess we can discuss what version we keep during the meeting (for when I will need to do the merge of master into my PR).